torch/distributions/gamma.py

```diff
@@ -65,6 +65,8 @@ def rsample(self, sample_shape=torch.Size()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
+        if not self._validate_args and isinstance(value, Number):
+            value = torch.tensor(value)
```

Comment:
You can do `torch.as_tensor` to have tensors go untouched, and have numbers wrapped automatically (plus e.g. numpy arrays will get accepted too).

Reply:
Could you please clarify if I should remove the if condition and switch to `torch.as_tensor` or should I remove these lines altogether in HalfCauchy and Gamma in support of the fact that official support is only for tensors and passing floats would generate an error? (following the comments to the previous review)
Also, the change in Uniform did not involve explicit conversion to tensor. So perhaps that can remain as it is?

Reply:
@vbsinha The changes in uniform are good to me, since they are just a convenient rewriting that happen to support floats.

@apaszke What dtype is picked with `torch.as_tensor` when none are specified? What convention do we have for such implicit conversion if we have one?

Reply:
It seems dangerous to me to have happy-path behavior depend on `self._validate_args`. I'd at least remove the `not self._validate_args and` part.

Reply:
@vincentqb IIRC `as_tensor` will use the default tensor type for floating point numbers (so `float` unless someone has changed it to `double`) and `int64` for everything else.